### PR TITLE
code cleanup fnc_blending

### PR DESCRIPTION
**Pull request by commy2**
**When merged this pull request will:**
- cont. from #3873
- also fix this bug for mines and satchel charges ("Put")
- private

### CODE REVIEW

**Comment by PabstMirror** [comment]
```
["firedPlayer", DFUNC(blending)] call EFUNC(common,addEventHandler);
["firedPlayerVehicle", DFUNC(blending)] call EFUNC(common,addEventHandler);
```

So some stuff like `currentVisionMode _unit` and `currentVisionMode weaponAccessories "HMG_M2"` will be for the vehicle instead of the player and might be nil/weird.

**Comment by commy2** [comment]
Good catch. `_unit` really is a bit confusing for this. What is `_vehicle` exactly?

File: addons/nightvision/functions/fnc_blending.sqf [hunk 1]
```diff
@@ -18,44 +18,33 @@
...
-_fnc_isTracer = {
-    private ["_indexShot", "_lastRoundsTracer", "_tracersEvery"];
+private _visibleFire = getNumber (configFile >> "CfgAmmo" >> _ammo >> "visibleFire");
+private _visibleFireTime = getNumber (configFile >> "CfgAmmo" >> _ammo >> "visibleFireTime");
 
+private _fnc_isTracer = {
     if (getNumber (configFile >> "CfgAmmo" >> _ammo >> "nvgOnly") > 0) exitWith {false};
 
-    _indexShot = (_player ammo _weapon) + 1;
+    private _indexShot = (_gunner ammo _weapon) + 1;
```
Comment (commy2) [inline addons/nightvision/functions/fnc_blending.sqf]: tested and `_unit ammo _weapon` as well as `_gunner ammo _weapon` report the same correct value of the vehicle

**Comment by PabstMirror** [comment]
Now we get script errors when firing unmounted because `firedPlayerVehicle` only provides 7 elements (no _gunner).
```
[5536,62.746,2.026,"z\ace\addons\nightvision\functions\fnc_blending.sqf:19","firedEH:: _unit=A,  _weapon=arifle_MXC_F,  _muzzle=arifle_MXC_F,  _mode=Single,  _ammo=B_65x39_Caseless,  _magazine=30Rnd_65x39_caseless_mag,  _projectile=164014: tracer_red.p3d,  _vehicle=any,  _gunner=any  _turret=any"]
[5536,62.746,2.026,"z\ace\addons\nightvision\functions\fnc_blending.sqf:64","Player Shot Adjusting NVG Effect:  _visibleFire=any,  _visibleFireTime=any"]
Too many post effect params set
Error in expression <};

ace_nightvision_ppEffectMuzzleFlash ppEffectAdjust [1, 1, _visibleFire, [0, >
  Error position: <ppEffectAdjust [1, 1, _visibleFire, [0, >
  Error 6 elements provided, 32 expected
File z\ace\addons\nightvision\functions\fnc_blending.sqf, line 52
```

**Comment by commy2** [comment]
should be fixed.

**Pull request merged**